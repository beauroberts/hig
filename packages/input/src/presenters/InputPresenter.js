import React from "react";
import PropTypes from "prop-types";
import { css } from "emotion";

import { ThemeContext } from "../../../themes/build";

import stylesheet from "./InputPresenter.stylesheet";
import { availableTypes } from "../constants";

export default function InputPresenter({
  disabled,
  hasFocus,
  hasHover,
  onBlur,
  onChange,
  onFocus,
  onInput,
  onMouseEnter,
  onMouseLeave,
  value,
  type
}) {
  return (
    <ThemeContext.Consumer>
      {({ themeData }) => {
        const styles = stylesheet(
          { isDisabled: disabled, hasFocus, hasHover, type },
          themeData
        );

        return (
          <input
            className={css(styles.input)}
            disabled={disabled}
            onBlur={onBlur}
            onChange={onChange}
            onFocus={onFocus}
            onInput={onInput}
            onMouseEnter={onMouseEnter}
            onMouseLeave={onMouseLeave}
            value={value}
          />
        );
      }}
    </ThemeContext.Consumer>
  );
}

InputPresenter.propTypes = {
  disabled: PropTypes.bool,
  hasFocus: PropTypes.bool,
  hasHover: PropTypes.bool,
  onBlur: PropTypes.func,
  onChange: PropTypes.func,
  onFocus: PropTypes.func,
  onInput: PropTypes.func,
  onMouseEnter: PropTypes.func,
  onMouseLeave: PropTypes.func,
  value: PropTypes.string,
  type: PropTypes.oneOf(availableTypes)
};
