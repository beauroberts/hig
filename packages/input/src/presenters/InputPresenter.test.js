import React from "react";
import renderer from "react-test-renderer";
import { mount } from "enzyme";

import InputPresenter from "./InputPresenter";

describe("InputPresenter", () => {
  it("renders by default", () => {
    expect(renderer.create(<InputPresenter />).toJSON()).toMatchSnapshot();
  });

  it("renders with hover", () => {
    expect(
      renderer.create(<InputPresenter hasHover />).toJSON()
    ).toMatchSnapshot();
  });

  it("renders with focus", () => {
    expect(
      renderer.create(<InputPresenter hasFocus />).toJSON()
    ).toMatchSnapshot();
  });

  it("renders when disabled", () => {
    expect(
      renderer.create(<InputPresenter isDisabled />).toJSON()
    ).toMatchSnapshot();
  });

  it("sets props on the input", () => {
    const onInputHandler = () => {};
    const onChangeHandler = () => {};
    const onFocusHandler = () => {};
    const onBlurHandler = () => {};
    const onMouseEnterHandler = () => {};
    const onMouseLeaveHandler = () => {};
    const wrapper = mount(
      <InputPresenter
        onChange={onChangeHandler}
        onFocus={onFocusHandler}
        onBlur={onBlurHandler}
        onInput={onInputHandler}
        onMouseEnter={onMouseEnterHandler}
        onMouseLeave={onMouseLeaveHandler}
      />
    );

    const inputProps = wrapper.find("input").props();
    expect(inputProps.onChange).toEqual(onChangeHandler);
    expect(inputProps.onFocus).toEqual(onFocusHandler);
    expect(inputProps.onBlur).toEqual(onBlurHandler);
    expect(inputProps.onMouseEnter).toEqual(onMouseEnterHandler);
    expect(inputProps.onMouseLeave).toEqual(onMouseLeaveHandler);
    expect(inputProps.onInput).toEqual(onInputHandler);
  });
});
