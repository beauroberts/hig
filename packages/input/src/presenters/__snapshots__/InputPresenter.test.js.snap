// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputPresenter renders by default 1`] = `
.emotion-0 {
  background-color: transparent;
  box-sizing: border-box;
  border: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  outline: none;
  font-size: 14px;
  font-family: ArtifaktElement,sans-serif;
  font-weight: 600;
  line-height: 1.4;
  width: 100%;
  color: #3c3c3c;
  -webkit-transition-property: color;
  transition-property: color;
  -webkit-transition-duration: 0.3s;
  transition-duration: 0.3s;
}

<input
  className="emotion-0"
  disabled={undefined}
  onBlur={undefined}
  onChange={undefined}
  onFocus={undefined}
  onInput={undefined}
  onMouseEnter={undefined}
  onMouseLeave={undefined}
  value={undefined}
/>
`;

exports[`InputPresenter renders when disabled 1`] = `
.emotion-0 {
  background-color: transparent;
  box-sizing: border-box;
  border: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  outline: none;
  font-size: 14px;
  font-family: ArtifaktElement,sans-serif;
  font-weight: 600;
  line-height: 1.4;
  width: 100%;
  color: #3c3c3c;
  -webkit-transition-property: color;
  transition-property: color;
  -webkit-transition-duration: 0.3s;
  transition-duration: 0.3s;
}

<input
  className="emotion-0"
  disabled={undefined}
  onBlur={undefined}
  onChange={undefined}
  onFocus={undefined}
  onInput={undefined}
  onMouseEnter={undefined}
  onMouseLeave={undefined}
  value={undefined}
/>
`;

exports[`InputPresenter renders with focus 1`] = `
.emotion-0 {
  background-color: transparent;
  box-sizing: border-box;
  border: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  outline: none;
  font-size: 14px;
  font-family: ArtifaktElement,sans-serif;
  font-weight: 600;
  line-height: 1.4;
  width: 100%;
  color: #3c3c3c;
  -webkit-transition-property: color;
  transition-property: color;
  -webkit-transition-duration: 0.1s;
  transition-duration: 0.1s;
}

<input
  className="emotion-0"
  disabled={undefined}
  onBlur={undefined}
  onChange={undefined}
  onFocus={undefined}
  onInput={undefined}
  onMouseEnter={undefined}
  onMouseLeave={undefined}
  value={undefined}
/>
`;

exports[`InputPresenter renders with hover 1`] = `
.emotion-0 {
  background-color: transparent;
  box-sizing: border-box;
  border: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  outline: none;
  font-size: 14px;
  font-family: ArtifaktElement,sans-serif;
  font-weight: 600;
  line-height: 1.4;
  width: 100%;
  color: #3c3c3c;
  -webkit-transition-property: color;
  transition-property: color;
  -webkit-transition-duration: 0.1s;
  transition-duration: 0.1s;
}

<input
  className="emotion-0"
  disabled={undefined}
  onBlur={undefined}
  onChange={undefined}
  onFocus={undefined}
  onInput={undefined}
  onMouseEnter={undefined}
  onMouseLeave={undefined}
  value={undefined}
/>
`;
